src/main/java/entity/BaseEntity.java

```diff
@@ -1,8 +1,16 @@
 package entity;
 
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.MappedSuperclass;
+
 import java.util.UUID;
 
+@MappedSuperclass
 public abstract class BaseEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
     protected UUID id;
 
     protected BaseEntity() {
```

Comment:
when you use `@GeneratedValue(strategy = GenerationType.AUTO)` there is no need to set `id` in constructor, it will be generated on db level:
```java
protected BaseEntity() {}
```